chained/water_impl.go

```diff
@@ -0,0 +1,47 @@
+package chained
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"net"
+
+	"github.com/getlantern/common/config"
+	"github.com/getlantern/flashlight/v7/ops"
+	"github.com/refraction-networking/water"
+	_ "github.com/refraction-networking/water/transport/v0"
+)
+
+type waterImpl struct {
+	config         *water.Config
+	raddr          string
+	reportDialCore reportDialCoreFn
+	nopCloser
+}
+
+func newWaterImpl(addr string, pc *config.ProxyConfig, reportDialCore reportDialCoreFn) (*waterImpl, error) {
+	b64WASM := ptSetting(pc, "water_wasm")
+	wasm, err := base64.StdEncoding.DecodeString(b64WASM)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode water wasm: %w", err)
+	}
+
+	return &waterImpl{
+		raddr: addr,
+		config: &water.Config{
```

Comment:
I think we want to provide a logger (using `water.Config.OverrideLogger`).  This should use `golog`, like the rest of flashlight's logs.

Reply:
hmmmm in this case golog isn't accepted by OverrideLogger because it expects a `slog.Logger` and the `golog` doesn't implement that, I'll try to do this but Idk if it'll work (probably requires `TRACE` env var set to true):
```go
	return &waterImpl{
		raddr: addr,
		config: &water.Config{
			TransportModuleBin: wasm,
			OverrideLogger:     slog.New(slog.NewTextHandler(log.TraceOut(), nil)),
		},
		reportDialCore: reportDialCore,
	}, nil

```

Reply:
Yeah, you'll need to wrap `golog` to get it to fit.  Sorry if that was unclear.  Looks like you will need to provide a custom [slog.Handler](https://pkg.go.dev/log/slog#Handler).  Admittedly, that looks like more work than I originally anticipated.  I think it still looks reasonable though?  Could you give it a go and let me know if you run into any major issues?  I don't think we want to just have logs dumped to stdout or a framework outside of our normal logging framework.

Reply:
I'll add some tests later after adding suggestions from http-proxy and lantern